plugin-server/src/main/ingestion-queues/ingest-event.ts

```diff
@@ -48,6 +53,20 @@ export async function ingestEvent(
                 timeoutMessage: `After 30 seconds still running ${isSnapshot ? 'onSnapshot' : 'onEvent'}`,
             }),
         ])
+
+        if (actionMatches.length > 0) {
+            await Promise.all(
+                actionMatches.map((actionMatch) =>
+                    runInstrumentedFunction({
+                        server,
+                        event: processedEvent!,
+                        func: (event) => workerMethods.onAction(actionMatch, event),
+                        statsKey: `kafka_queue.on_action`,
+                        timeoutMessage: 'After 30 seconds still running onAction',
+                    })
+                )
+            )
+        }
```

Comment:
I wonder if changing the flow from:

```
sequential(
  parallel(IngestEvent, onEvent),
  onAction,
)
```

To:

```
parallel(
  sequential(IngestEvent, onAction), 
  onEvent
)
```

makes any sense? 

Probably not, as I expect `ingestEvent` to take much much longer 99.99% of the time than `onEvent`, which normally uses a buffer.